src/integrations/prefect-azure/pyproject.toml

```diff
@@ -0,0 +1,108 @@
+[build-system]
+requires = ["setuptools>=45", "wheel", "setuptools_scm>=6.2"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "prefect-azure"
+description = "Prefect integrations with Microsoft Azure services"
+readme = "README.md"
+requires-python = ">=3.8"
+license = {text = "Apache License 2.0"}
+keywords = ["prefect"]
+authors = [
+  {name = "Prefect Technologies, Inc.", email = "help@prefect.io"}
+]
+classifiers = [
+  "Natural Language :: English",
+  "Intended Audience :: Developers",
+  "Intended Audience :: System Administrators",
+  "License :: OSI Approved :: Apache Software License",
+  "Programming Language :: Python :: 3 :: Only",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Topic :: Software Development :: Libraries",
+]
+dependencies = [
+  "aiohttp",
+  "azure_identity>=1.10",
+  "azure_mgmt_containerinstance>=10.0",
+  "azure-mgmt-resource>=21.2",
+  "prefect>=2.14.10",
+]
+dynamic = ["version"]
+
+[project.optional-dependencies]
+blob_storage = ["azure-storage-blob"]
+cosmos_db = ["azure-cosmos"]
+ml_datastore = ["azureml-core"]
+all_extras = [
+  "azure-cosmos",
+  "azure-storage-blob",
+  "azureml-core",
+]
+dev = [
+  "aiohttp",
+  "azure-cosmos",
+  "azure-storage-blob",
+  "azureml-core",
+  "black",
+  "coverage",
+  "flake8",
+  "interrogate",
+  "isort",
+  "mkdocs-gen-files",
+  "mkdocs-material",
+  "mkdocs",
+  "mkdocstrings[python]",
+  "mock; python_version < '3.8'",
+  "mypy",
+  "pillow",
+  "pre-commit",
+  "pytest-asyncio",
+  "pytest",
+  "pytest-xdist",
+]
+
+[project.urls]
+Homepage = "https://github.com/PrefectHQ/prefect-azure"
+
+[project.entry-points."prefect.collections"]
+prefect_azure = "prefect_azure"
+
+[tool.setuptools_scm]
+version_file = "prefect_azure/_version.py"
+root = "../../.."
+tag_regex = "^prefect-azure-(?P<version>\\d+\\.\\d+\\.\\d+)$"
+fallback_version = "0.0.0"
```

Comment:
These settings ensure that only tags prefixed with `prefect-azure-` will be used to determine the `prefect-azure` version.